packages/monitoring/src/types/index.ts

```diff
@@ -1,40 +1,36 @@
+import { KeypairType } from "@polkadot/util-crypto/types";
 /**
  * Represents a basic service interface.
+ * @template P - The type of the parameter object used for the update method.
  */
-interface IServiceBase {
+interface IServiceBase<P> {
   /**
-   * Returns the name of the service.
-   * @returns {string} The service name.
+   * The name of the service.
    */
-  serviceName: () => string;
+  name: string;
 
   /**
-   * Returns the URL of the service.
-   * @returns {string} The service URL.
+   * The URL of the service.
    */
-  serviceUrl: () => string;
-}
+  url: string;
 
-/**
- * Represents a handler for disconnecting a service.
- */
-export interface IDisconnectHandler {
   /**
-   * Performs the disconnection from the service.
-   * @returns {Promise<void>} A promise that resolves when the disconnection is successful.
+   * Updates the service with the provided parameters.
+   * @param {P} param - The parameter object with specific keys, should be specified on class.
    */
-  disconnect: () => Promise<void>;
+  update(param: P): void;
 }
```

Comment:
do we still need this update method?

Reply:
I guess I should replace it with normal setter, and use it [here](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3134/files#diff-0741e94a896c535b6c4a88b69466d0ab9bbd819c6367c57aab6ab85528051e32R127-R132), if with the reachable url